tests/hooks/test_model_hooks.py

```diff
@@ -0,0 +1,469 @@
+            "target_host",
+            "target_name",
+            "target_schema",
+            "target_threads",
+            "target_type",
+            "target_user",
+            "target_pass",
+            "run_started_at",
+            "invocation_id",
+        ]
+        field_list = ", ".join(['"{}"'.format(f) for f in fields])
+        query = f"select {field_list} from {SOURCE}.{project.test_schema}.on_model_hook where test_state = '{state}'"
+
+        vals = project.run_sql(query, fetch="all")
+        assert len(vals) != 0, "nothing inserted into hooks table"
+        assert len(vals) >= count, "too few rows in hooks table"
+        assert len(vals) <= count, "too many rows in hooks table"
+        return [{k: v for k, v in zip(fields, val)} for val in vals]
+
+    def check_hooks(self, state, project, host, count=1):
+        ctxs = self.get_ctx_vars(state, count=count, project=project)
+        for ctx in ctxs:
+            assert ctx["test_state"] == state
+            assert ctx["target_dbname"] is None
+            assert ctx["target_host"] is None
+            assert ctx["target_name"] == "default"
+            assert ctx["target_schema"] == f"{SOURCE}.{project.test_schema}"
+            assert ctx["target_threads"] == 1
+            assert ctx["target_type"] == "dremio"
+            assert ctx["target_user"] == os.getenv("DREMIO_CLOUD_USERNAME")
+            assert ctx["target_pass"] is None
+
+            assert (
+                ctx["run_started_at"] is not None and len(ctx["run_started_at"]) > 0
+            ), "run_started_at was not set"
+            assert (
+                ctx["invocation_id"] is not None and len(ctx["invocation_id"]) > 0
+            ), "invocation_id was not set"
+
+
+class TestPrePostModelHooksDremio(BaseTestPrePost):
+    @pytest.fixture(scope="class")
+    def project_config_update(self):
+        return {
+            "models": {
+                "test": {
+                    "pre-hook": [
+                        # inside transaction (runs second)
+                        MODEL_PRE_HOOK,
+                        # outside transaction (runs first)
+                        {
+                            "sql": "vacuum {{ this.schema }}.on_model_hook",
+                            "transaction": False,
+                        },
+                    ],
+                    "post-hook": [
+                        # outside transaction (runs second)
+                        {
+                            "sql": "vacuum {{ this.schema }}.on_model_hook",
+                            "transaction": False,
+                        },
+                        # inside transaction (runs first)
+                        MODEL_POST_HOOK,
+                    ],
+                }
+            }
+        }
+
+    @pytest.fixture(scope="class")
+    def models(self):
+        return {"hooks.sql": models__hooks}
+
+    def test_pre_and_post_run_hooks(self, project, dbt_profile_target):
```

Comment:
For tests that you copy over, I _believe_ you don't need to copy the test if nothing is being overwritten in the test definition itself. As long as you override the fixtures, the inherited class will use the overridden fixture. If this is the case, you can remove a lot of the actual test functions from these classes. Correct me if I'm wrong.

Reply:
This test inherits from the local BaseTestPrePost class which has been modified. If I inherited from the TestPrePostModelHooks test class from dbt-core, it would not use the modified BaseTestPrePost. Because we cannot inherit, we need to define all of these test functions.

I just tested it and can confirm that inheriting from TestprePostModelHooks would lead the test to fail even with modified fixtures.